proposed/extended-coding-style-guide.md

```diff
@@ -122,22 +125,77 @@ declaration.
 
 There MUST be one `use` keyword per declaration.
 
-There MUST be one blank line after the `use` block.
+When using multiple classes, functions or constants within one namespace, you
+MUST group use statements within one namespace.
+
+Use statements MUST be in a block according to the entity (class, function or group)
+which is being grouped. Within each block there MUST be no blank lines. If a block
+has multiple lines there MUST be a blank line before the first line and a blank line
+after the last line.
+
+Classes, functions or constants grouped together into a single line must be listed
+alphabetically.
+
+The groups MUST be ordered such that classes are first, followed by functions and
+then constants.
 
 For example:
 
 ```php
 <?php
+declare(strict_types=1);
+
 namespace Vendor\Package;
 
-use FooClass;
-use BarClass as Bar;
-use OtherVendor\OtherPackage\BazClass;
+use Vendor\Package\{ClassA as A, ClassB, ClassC as C};
```

Comment:
If you have enough for it to go only multiple lines then it's much more readable like this still I think. And it's not necessarily just four, after all line soft limit is 120 characters. It only makes things easier/readability easier if it's a small number for one, and when it's all one one line secondly I think.